AutorunsToWinEventLog/AutorunsToWinEventLog.ps1

```diff
@@ -32,3 +32,44 @@ Foreach ($item in $autorunsArray) {
   $item = $(Write-Output $item  | Out-String -Width 1000)
   Write-EventLog -LogName Autoruns -Source AutorunsToWinEventLog -EntryType Information -EventId 1 -Message $item
 }
+
+# Collect principals in interesting local groups (Administrators, Remote Desktop Users, and DCOM Users)
+# Requires PowerShell 5.1 due to usage of Get-NetLocalGroup and Get-LocalGroupMember
+
+# Get the FQDN of the current user's domain. Todo: update this method to support foreign security principals in local groups.
+$DomainFQDN = $ENV:USERDNSDOMAIN
+
+$ComputerName = (Get-WmiObject win32_computersystem).DNSHostName + "." + (Get-WmiObject -class win32_computersystem).Domain
+
+$LocalGroups = Get-LocalGroup | ?{$_.SID -Match "S-1-5-32-555" -Or $_.SID -Match "S-1-5-32-544" -Or $_.SID -Match
+ "S-1-5-32-562"}
+
+$LocalGroups | % {
+
+    $GroupName = $_
+
+    Get-LocalGroupMember -Name $GroupName | ?{
+        $_.PrincipalSource -Match "ActiveDirectory"
+    } | % {
+
+        $principalName = $_.Name.Split("\")[1] + "@" + $DomainFQDN
+
+        $Member = New-Object PSObject
+        $Member | Add-Member Noteproperty 'ComputerName' $ComputerName
+        $Member | Add-Member Noteproperty 'GroupName' $GroupName
+        $Member | Add-Member Noteproperty 'PrincipalType' $_.ObjectClass
+        $Member | Add-Member Noteproperty 'PrincipalName' $principalname
+
+        $Data = @"
+
+ComputerName: $($Member.ComputerName)
+GroupName: $($Member.GroupName)
+PrincipalType: $($Member.PrincipalType)
+PrincipalName: $($Member.PrincipalName)
+"@
+
+        Write-EventLog -LogName LocalGroupMembers -Source LocalGroupMembersScript -EntryType Information -EventId 1 -Message $Data
```

Comment:
LogName should be `Autoruns`
Source should be `AutorunsToWinEventLog`
EventId should be `2`